geode-wan/src/test/java/org/apache/geode/internal/cache/wan/WANTestBase.java

```diff
@@ -3171,6 +3182,35 @@ public static String displayQueueContent(final RegionQueue queue) {
     return null;
   }
 
+  public static String displaySerialQueueContent(final AbstractGatewaySender sender) {
+    StringBuilder message = new StringBuilder();
+    message.append("Is Primary: ").append(sender.isPrimary()).append(", ").append("Queue Size: ")
+        .append(sender.getEventQueueSize());
+
+    if (sender.getQueues() != null) {
+      message.append(", ").append("Queue Count: ").append(sender.getQueues().size());
+      Stream<Object> stream = sender.getQueues().stream()
+          .map(regionQueue -> ((SerialGatewaySenderQueue) regionQueue).displayContent());
+
+      List<Object> list = stream.collect(Collectors.toList());
+      message.append(", ").append("Keys: ").append(list.toString());
+    }
+
+    AbstractGatewaySenderEventProcessor abstractProcessor = sender.getEventProcessor();
+    if (abstractProcessor == null) {
+      message.append(", ").append("Null Event Processor: ");
+    }
+    if (sender.isPrimary()) {
+      message.append("\n").append("Unprocessed Tokens: ")
+          .append(abstractProcessor.printUnprocessedTokens());
+    } else {
+      message.append("\n").append("Unprocessed Events: ")
+          .append(abstractProcessor.printUnprocessedEvents());
+    }
```

Comment:
unprocessedEvents and unprocessedTokens are both only on the secondary. They are updated in the basicHandlePrimaryEvent (called by SerialSecondaryGatewayListener) and basicHandleSecondaryEvent (called by enqueueEvent).

If the order is: primary sees event first, then secondary, the event id will be put into and removed from the unprocessedTokens.
If the order is: secondary sees event first, then primary, the event id will be put into and removed from the unprocessedEvents.

Reply:
I will fix it